torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -192,6 +192,20 @@ class CPUOffload:
     offload_params: bool = False
 
 
+@dataclass
+class ForwardPrefetch:
+    """
+    Forward full parameters prefetch config.
+    prefetch_full_params: Initialize all-gather to rebuild full parameters of
+                          the layer right before the forward pass computation
+                          of the previous layer. This approach enables communication
+                          and computation overlap in the forward pass. It might increase
+                          the peak memory usage as the prefetched full parameters will be
+                          kept in the GPU memory until next layer's forward computation is done.
+    """
+    prefetch_full_params: bool = False
+
```

Comment:
maybe we can just make it as bool flag instead of data class, because there is no planned options in the future?

Reply:
Yes that is cleaner and is also what I used at first, the reason I added a `ForwardPrefetch` class is to make it consistent for test_name_mapping, see https://github.com/pytorch/pytorch/pull/78841/files#diff-bc070ccbc574b49b42259aab3e85253a5b5dae8f18e82343f6f85fb44c998b59R58 
If making it a bool flag, I can't think of a good way to automatically general test names based on the test_name_mapping, any suggestions would be appreciated

Reply:
Done